readme.md

```diff
@@ -62,6 +62,61 @@ const run = async () => {
 })();
 ```
 
+With the `onFailedAttempt` option and optional `context`:
+
+```js
+const pRetry = require('p-retry');
+
+class AuthenticatedPoller {
+    constructor (oauthToken) {
+        this.token = oauthToken;
+    }
+
+    failHandler (error, context) {
+        console.log(JSON.stringify(error));
+        console.log(`Attempt ${error.attemptNumber} failed. There are ${error.retriesLeft} retries left.`);
+
+        if (error.response.statusCode === 401 && error.response.error === 'OAuthTokenExpired') {
+            console.log(`OAuth token ${poller.token} has expired, refreshing token`);
+            context.refreshOAuthToken();
+        }
+    }
+
+    startPolling() {
+        pRetry(() => this.callAPI(this.token), {
+            onFailedAttempt: this.failHandler,
```

Comment:
Just use `Function#bind()` and you won't need the `context` option:

```suggestion
            onFailedAttempt: this.failHandler.bind(this),
```

You can then use `this` in the `failHandler()` method.

Reply:
That's a lot simpler than what I was trying to do : )

Function#bind() accomplished exactly what I needed - Thanks for help!